pkg/storage/filesystem/migrate.go

```diff
@@ -0,0 +1,76 @@
+package filesystem
+
+import (
+	"context"
+	"io/ioutil"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	zeros [512]byte
+)
+
+func wipe(ctx context.Context, device *Device, exe executer) error {
+	//we first make an msdos parition, because we know it's 512 bytes
+	//creating the msdos partition table also makes us validate that this
+	//is indeed a valid block device
+	if _, err := exe.run(ctx, "parted", "-s", device.Path, "mktable", "msdos"); err != nil {
```

Comment:
I don't get this. Why do we re-created a partition table already  and not just write zeros on the head of the disk to remove any existing filesystems?
Storaged initialization does all this automatically when he detects a disk that has no filesystem

Reply:
I am avoiding issues that can get triggered by having an unknown partition table on the disk. May be it's not gpt, so i can't just write gpt size bytes to disk. What if it's a different partition table that is more dynamic, and may be writing to the first 512 bytes won't affect it. 

So instead, I make sure we do msdos partition which is always 512 bytes on the fist sector. And then wipe it out with zeros. 

Also note that, using parted also works as a sort of validation that this is indeed a block device.

I don't see which part of the code you pointing at that storaged does that automatically!

Reply:
alright I get what you do here. So you first create a partition of known size, so you know how much you need to zeroed.
But if this is the size of the partition table you're afraid of. then why don't we just zeroed a big portion of the disk ? few MiB ?

Reply:
I did a quick test to erase a gpt partition by only writing zeros to the 512 bytes header, and it seems it works fine. I will just make sure by writing zeros of 1MB

I will also try to find a way to call both `sync` and `partprope` via systemcalls instead of calling binaries.

Reply:
can't we avoid the need to call sync if we open the file in direct IO mode ?